round-based/src/state_machine/runtime.rs

```diff
@@ -0,0 +1,47 @@
+use core::task::{ready, Poll};
+
+/// State machine runtime
+pub struct Runtime<M> {
+    shared_state: super::shared_state::SharedStateRef<M>,
+}
+
+impl<M> Runtime<M> {
+    pub(super) fn new(shared_state: super::shared_state::SharedStateRef<M>) -> Self {
+        Self { shared_state }
+    }
+}
+
+impl<'a, M> crate::runtime::AsyncRuntime for Runtime<M> {
```

Comment:
Unnecessary lifetime parameter, here and below. It seems you we don't run clippy on this repo?

Reply:
I see in CI we do, but it gives warnings that we ignore, plus my local clippy with `--all-features -- -D clippy::all` gives more diagnostics

Reply:
How does it even compiles...

Reply:
Fixed